src/inject/dynamic-theme/css-rules.ts

```diff
@@ -1,32 +1,53 @@
 import {forEach} from '../../utils/array';
 import {isSafari} from '../../utils/platform';
 import {parseURL, getAbsoluteURL} from '../../utils/url';
+import {addDOMCompleteListener, isCompleteDomReady} from '../utils/dom';
 import {logWarn} from '../utils/log';
 
-export function iterateCSSRules(rules: CSSRuleList, iterate: (rule: CSSStyleRule) => void) {
-    forEach(rules, (rule) => {
-        if (rule instanceof CSSMediaRule) {
-            const media = Array.from(rule.media);
-            const isScreenOrAll = media.some((m) => m.startsWith('screen') || m.startsWith('all'));
-            const isPrintOrSpeech = media.some((m) => m.startsWith('print') || m.startsWith('speech'));
+export async function iterateCSSRules(rules: CSSRuleList, iterate: (rule: CSSStyleRule) => void) {
```

Comment:
As far as I understand, you should leave the function `iterateCSSRules` synchronous. I don't see `await iterateCSSRules` anywhere

Reply:
So, here is where the "fix" actually relies. We are adding a listener inside this function, to wait for when the `document.readyState === 'complete'`, but this only happens when we have a LinkStyle that we cannot access it's styles, this is because it's still loading.

Now you will think, yeah so? just have the callback execute the `iterate` function.

Correct, that should do the job, however. in `stylesheet-modifer.ts` we would have `iterateCSSRules(rules, (rule) => {` and after the last rule is being iterated(so not the ones that we are waiting for in the callback for `complete` readyState). And it would use the modRules to modify the `darkreader--sync` sheet. So now when it happens that the callback would be fired, it would add the modRules, but because due to the fact that it treated as a synchronous function it won't have any effect because the `buildOverride()` etc is already fired and executed. 

It's a complicated issue that can be paired with "race conditions", because you will execute a callback after the `modifySheet` function has already finished modifying the sheet. But when we will treat is as an asynchronous function, it will await the last `iterate` of the loaded CSSImportRule and thus will be included in the whole modifying process.

Reply:
I tried observing and here is what I think:

Making this function asynchronous, and using awaits makes some parts of the code being executed after another in an unpredictable way. This can cause problems when reading variables or other cases. I wish the execution flow will be straightforward.

I found out that the problem actually happens (unstyled content on docs.python.org) when error happens in
```typescript
try {
                        iterateCSSRules(rule.styleSheet.cssRules, iterate);
                    } catch (err) {
                        logWarn(err);
                    }
```

So I think what you can do here is revert the functions to being synchronous, and when error happens with reading rules of CSSImportRule, rerun building the sheet override when loading ends (ideally loading of the link, not a document). One of the ways to achieve this is:
```typescript
        function buildOverrides() {
            const force = forceRenderStyle;
            forceRenderStyle = false;
            sheetModifier.modifySheet({
                ...
            });
            isOverrideEmpty = syncStyle.sheet.cssRules.length === 0;
            if (sheetModifier.hasNotLoadedImports()) {
                // "update" function schedules rebuilding the style
                // ideally to wait for link loading, because some sites put links any time,
                // but it can be complicated, so waiting for document completion can do the trick
                addReadyStateCompleteListener(() => update());
            }
        }
```